### PR TITLE
More descriptive message for ``use-implicit-booleaness-not-comparison``

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Follow-up to #7723 in order to make #6870 more reviewable.

### CODE REVIEW

**Comment by mbyrnepr2** [comment]
Are you concerned about the first two Sentry primer messages @Pierre-Sassoulas or can we ignore/they are unrelated?

File: pylint/checkers/refactoring/implicit_booleaness_checker.py [hunk 1]
```diff
@@ -197,23 +196,32 @@ def _check_use_implicit_booleaness_not_comparison(
...
-        if isinstance(literal_node, nodes.List):
-            collection_literal = "[]"
-        if isinstance(literal_node, nodes.Tuple):
-            collection_literal = "()"
+        description = self._get_node_description(literal_node)
+        collection_literal = {
+            "list": "[]",
+            "tuple": "()",
+            "dict": "{}",
+        }.get(description, "iterable")
```
Comment (mbyrnepr2) [inline pylint/checkers/refactoring/implicit_booleaness_checker.py:216]: I wonder would it be best to factor this logic out and into the `_get_node_description` method? It is only used in this one place.

Comment (Pierre-Sassoulas) [reply pylint/checkers/refactoring/implicit_booleaness_checker.py:216]: You're right, it does not make a lot of sense right now but I'm going to use it use-implicit-booleaness-not-len later :)

**Review by mbyrnepr2 — APPROVED** [review]
You are on 🔥🔥 @Pierre-Sassoulas

**Pull request merged**